### PR TITLE
Check deployability

### PR DESCRIPTION
**Pull request by KevinK-9**
## Changes Proposed:

- add functionality for checking if the queue is full

## Issues Addressed:

- Closes

## Tests Performed:

- added test case to deployments.spec.ts

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test PRs
1) First fetch the Pull-Request from the main repo, replacing `xx` with the ID of the PR:\
`git fetch git@github.com:eclipsesource/cdtcloud-deploymentserver.git pull/xx/head:pr_xx`
2) Checkout the Pull-Request, again replacing `xx` with the PR-ID:\
`git checkout pr_xx`
3) Perform the necessary tests for the PR

### CODE REVIEW

File: packages/deployment-server/src/deployments/routes.ts [hunk 1]
```diff
@@ -56,6 +56,10 @@ export default function deploymentRequestsRoutes (router: Router): void {
           return res.sendStatus(503)
```
Comment (WoH) [inline packages/deployment-server/src/deployments/routes.ts:56]: Add a similar structure

File: packages/deployment-server/test/deployment.spec.ts [hunk 2]
```diff
@@ -235,3 +235,132 @@ test('Closes the i/o stream after a deployment request is cancelled', async (t)
...
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      deviceTypeId: typeId,
+      artifactUri
+    })
+  })
+
+  t.equal(response.status, 502)
```
Comment (WoH) [inline packages/deployment-server/test/deployment.spec.ts:365]: match the response body

File: packages/deployment-server/test/deployment.spec.ts [hunk 3]
```diff
@@ -235,3 +235,132 @@ test('Closes the i/o stream after a deployment request is cancelled', async (t)
...
+  const formData = new FormData()
+  const file = new File([arraybuffer], 'artifact', { type: 'text/plain' })
+  formData.append('file', file)
+
+  const artifactResponse = await fetch(`${baseUrl}/deployment-artifacts`, {
+    method: 'POST',
+    body: formData
+  })
+
+  const { artifactUri } = await artifactResponse.json() as {artifactUri: string}
```
Comment (WoH) [inline packages/deployment-server/test/deployment.spec.ts:253]: We don't need the actual artifact, we can submit any valid url

File: packages/deployment-server/test/deployment.spec.ts [hunk 4]
```diff
@@ -235,3 +235,132 @@ test('Closes the i/o stream after a deployment request is cancelled', async (t)
...
+        },
+        artifactUri,
+        id: /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/
+      }
+    })
+    socket.close()
+  }
+
+  // wait for the socket
+  await once(socket, 'open')
```
Comment (WoH) [inline packages/deployment-server/test/deployment.spec.ts:311]: No need to verify this again

File: packages/deployment-server/test/deployment.spec.ts [hunk 5]
```diff
@@ -235,3 +235,132 @@ test('Closes the i/o stream after a deployment request is cancelled', async (t)
...
+  await fetch(`${baseUrl}/deployments`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      deviceTypeId: typeId,
+      artifactUri
+    })
+  })
```
Comment (WoH) [inline packages/deployment-server/test/deployment.spec.ts:343]: Check that this one is a 200 with a deployment body

File: packages/deployment-server/test/deployment.spec.ts [hunk 6]
```diff
@@ -235,3 +235,132 @@ test('Closes the i/o stream after a deployment request is cancelled', async (t)
     }
   })
 })
+
+test('returns an error when queue is full', async (t) => {
```
Comment (WoH) [inline packages/deployment-server/test/deployment.spec.ts:239]: A better, more focussed approach: 
1) Create device (util/factory.ts). Update status to unavailable. Deploy fails
2) Create device (util/factory.ts). Add MAX_QUEUE_SIZE requests, all work
3) Create device (util/factory.ts). Add MAX_QUEUE_SIZE+ 1 requests, last one fails

**Review by theZasa — APPROVED** [review]
Merge for now, fixes later

**Pull request merged**